app/agent/runtime/qemu/qemu.service.test.ts

```diff
@@ -0,0 +1,387 @@
+import path from "path";
+
+import testImages from "../../block-registry/test-data/test_images.json";
+import { doesFileExist, execSshCmd, sleep } from "../../utils";
+
+import { EXPOSE_METHOD } from "~/agent/block-registry/registry.service";
+import { provideBlockRegistry } from "~/agent/block-registry/test-utils";
+import { Token } from "~/token";
+import { waitForPromises } from "~/utils.shared";
+
+const testCases = [
+  ["Alpine 3.14", testImages.testAlpine3_14],
+  ["Debian Bookworm", testImages.testDebianBookworm],
+  ["Debian Buster", testImages.testDebianBuster],
+  ["Ubuntu Focal", testImages.testUbuntuFocal],
+  ["Ubuntu Jammy", testImages.testUbuntuJammy],
+];
+
+test.concurrent.each(testCases)(`Boots and connects to %s`, async (_name, remoteTag) =>
+  provideBlockRegistry(async ({ injector, runId, brService }) => {
+    const osIm = await brService.loadImageFromRemoteRepo(remoteTag, "osIm");
+    const osCt = await brService.createContainer(osIm, "osCt");
+    const osB = await brService.expose(osCt, EXPOSE_METHOD.BLOCK_DEV);
```

Comment:
shouldn't you also hide the block device after you're finished with it?

Reply:
It's handled globally: https://github.com/hocus-dev/hocus/pull/93/files#diff-6728b317769e3b07f2f0926e1b27c3dadcd3958b1eb186a33b3cc8b3234f6047R162